core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -274,9 +275,16 @@ public MeterBinder newMeterBinder(MeterIdPrefix idPrefix) {
 
     @Override
     public String toString() {
+        final List<Endpoint> endpoints = endpoints().stream().limit(10).collect(Collectors.toList());
+        final List<Endpoint> candidates = delegate.endpoints().stream().limit(10)
+                                                  .collect(Collectors.toList());
```

Comment:
How about checking the size of `endpoints()` and `delegate.endpoints()` to avoid additional copies if it is less than or equal to 10?

Reply:
I've implemented a custom helper method since I wasn't able to find a java/guava api that allows for this behavior 😅